boards: nxp: imx95_evk_15x15: add i.MX 95 15x15 LPDDR4x EVK variant

asellaminxp opened the pull request:
Add initial support for i.MX 95 15x15 LPDDR4x EVK board. This board uses the i.MX 95 15x15 SoC that shares many similarities to the already supported i.MX 95 19x19 SoC used for the i.MX 95 19x19 LPDDR5 EVK.

This enables Zephyr to boot and run on the i.MX 95 15x15 EVK and provides a foundation for further peripheral enablement and application development.

Depends on:

- [ ] https://github.com/zephyrproject-rtos/hal_nxp/pull/553

asellaminxp committed:
boards: nxp: imx95_evk_15x15: add i.MX 95 15x15 LPDDR4x EVK variant

Add initial support for i.MX 95 15x15 LPDDR4x EVK board. This board
uses the i.MX 95 15x15 SoC that shares many similarities to the
already supported i.MX 95 19x19 SoC used for the i.MX 95 19x19
LPDDR5 EVK.

This enables Zephyr to boot and run on the i.MX 95 15x15 EVK and
provides a foundation for further peripheral enablement and
application development.

Signed-off-by: Aziz Sellami <aziz.sellami@nxp.com>

iuliana-prodan commented on boards/nxp/imx95_evk_15x15/Kconfig.imx95_evk_15x15:
```diff
@@ -0,0 +1,6 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_IMX95_EVK_15X15
```
Why do you need to add a new board for 15x15?
On a quick look all files you added here are the same as those from imx95_evk.

decsny replied:
if revisions are awkward because it's more of a parallel design than sequential, you can still make common files to share between the boards, instead of duplicating everything

mmahadevan108 commented:
@asellaminxp please re-visit this PR.

asellaminxp commented:
The build issue was due to missing pinmux nodes in the https://github.com/zephyrproject-rtos/hal_nxp/blob/master/dts/nxp/nxp_imx/mimx9596cvtxn-pinctrl.dtsi file, where `iomuxc_gpio_io35_gpio_io_bit_gpio5_io_bit15`, `iomuxc_gpio_io36_gpio_io_bit_gpio5_io_bit16` and `iomuxc_gpio_io37_gpio_io_bit_gpio5_io_bit17`.
A temporary fix would be to move the gpio pinmux definitions at the board level.

CC: @GhMarwen

JiafeiPan commented:
> The build issue was due to missing pinmux nodes in the https://github.com/zephyrproject-rtos/hal_nxp/blob/master/dts/nxp/nxp_imx/mimx9596cvtxn-pinctrl.dtsi file, where `iomuxc_gpio_io35_gpio_io_bit_gpio5_io_bit15`, `iomuxc_gpio_io36_gpio_io_bit_gpio5_io_bit16` and `iomuxc_gpio_io37_gpio_io_bit_gpio5_io_bit17`. A temporary fix would be to move the gpio pinmux definitions at the board level.
> 
> CC: @GhMarwen

Hi, @asellaminxp , pinmux is not board level, it should be SoC level. as 15x15 has different pinmux with 19x19, so it should have a standalone SoC dts,  as they have the same memory layout for all peripherals,  so could create a common dtsi common.dtsi  like to cover these common part, then in 19x19.dtsi and 15x15.dtsi will include common.dtsi and silicon specified pinmux dts nodes.

asellaminxp commented:
> > The build issue was due to missing pinmux nodes in the https://github.com/zephyrproject-rtos/hal_nxp/blob/master/dts/nxp/nxp_imx/mimx9596cvtxn-pinctrl.dtsi file, where `iomuxc_gpio_io35_gpio_io_bit_gpio5_io_bit15`, `iomuxc_gpio_io36_gpio_io_bit_gpio5_io_bit16` and `iomuxc_gpio_io37_gpio_io_bit_gpio5_io_bit17`. A temporary fix would be to move the gpio pinmux definitions at the board level.
> > CC: @GhMarwen
> 
> Hi, @asellaminxp , pinmux is not board level, it should be SoC level. as 15x15 has different pinmux with 19x19, so it should have a standalone SoC dts, as they have the same memory layout for all peripherals, so could create a common dtsi common.dtsi like to cover these common part, then in 19x19.dtsi and 15x15.dtsi will include common.dtsi and silicon specified pinmux dts nodes.

Indeed, I created the new DTSIs for 19x19 and 15x15, as suggested.

JiafeiPan commented on west.yml:
```diff
@@ -210,7 +210,7 @@ manifest:
         - hal
     - name: hal_nxp
-      revision: 111f568bda6f119cd896f38ae5843ecde92039bd
+      revision: 49ca79a20d932374a2c26e4f47153d17c038c0b5
```
revision: refs/pull/553/head

asellaminxp replied:
Done

JiafeiPan commented on boards/nxp/imx95_evk/imx95_evk_mimx9596_a55.dts:
```diff
@@ -6,7 +6,7 @@
 /dts-v1/;
 
-#include <nxp/nxp_mimx95_a55.dtsi>
+#include <nxp/nxp_mimx95_19x19_a55.dtsi>
```
I think this should be the first commit to add dts for 15x15, then the second one will add the board. so that it will not break bisect.

asellaminxp replied:
Done

JiafeiPan replied:
Suggest to refine the commit message like:
 dts: nxp: imx95-ca55: add dts for SoC 15x15 variant

As, i.MX 95 19x19 and i.MX 95 15x15 have different pinmux definitions, so keep
common part in nxp_mimx95_a55.dtsi, and define separate dts file for each variant,
they include common part and pinmux definitions.

asellaminxp committed:
dts: nxp: imx95-ca55: add dts for SoC 15x15 variant

As i.MX 95 19x19 and i.MX 95 15x15 have different pinmux definitions,
keep common part in nxp_mimx95_a55.dtsi, and define separate dts file
for each variants. These include the common part and their respective
pinmux definitions.

Signed-off-by: Aziz Sellami <aziz.sellami@nxp.com>

mmahadevan108 commented:
@iuliana-prodan , kindly revisit this and see if it addresses your block.
